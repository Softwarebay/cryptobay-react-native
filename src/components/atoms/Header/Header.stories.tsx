import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { Header } from './Header';

storiesOf('Atoms/Header', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Gradient Header', () => (
    <Header value={text('value', 'Convenient Transaction')} />
  ))
  .add('White Header', () => (
    <Header
      value={text('value', 'Millionsof people participate')}
      variant="white"
    />
  ));
