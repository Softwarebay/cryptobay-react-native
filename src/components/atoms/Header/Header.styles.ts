import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  header: {
    fontFamily: 'Poppins',
    lineHeight: 56,
  },
  gradient: {
    fontSize: 40,
    fontWeight: '300',
  },
  white: {
    fontSize: 42,
    fontWeight: '400',
    color: '#FFFFFF',
  },
  opacityZero: {
    opacity: 0,
  },
});
