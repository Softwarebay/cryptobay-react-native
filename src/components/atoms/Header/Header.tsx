import MaskedView from '@react-native-masked-view/masked-view';
import React from 'react';
import { Text, TextStyle } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import { styles } from './Header.styles';

type HeaderProps = {
  value: string;
  variant?: 'gradient' | 'white';
  style?: TextStyle | TextStyle[];
  testID?: string;
};

export const Header: React.FC<HeaderProps> = ({
  value,
  variant = 'gradient',
  style,
  testID,
}) => {
  if (variant === 'white') {
    return (
      <Text testID={testID} style={[styles.header, styles.white]}>
        {value}
      </Text>
    );
  }

  return (
    <MaskedView
      maskElement={
        <Text testID={testID} style={[styles.header, styles.gradient, style]}>
          {value}
        </Text>
      }>
      <LinearGradient
        colors={['#A9CDFF', '#72F6D1', '#A0ED8D', '#FED365', '#FAA49E']}
        start={{ x: 0, y: 0 }}
        end={{ x: 1, y: 0 }}>
        <Text
          style={[styles.header, styles.gradient, styles.opacityZero, style]}>
          {value}
        </Text>
      </LinearGradient>
    </MaskedView>
  );
};
