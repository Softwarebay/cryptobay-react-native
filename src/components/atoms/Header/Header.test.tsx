import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Header } from './Header';

describe('Header Atom', () => {
  test('Component Snapshot', () => {
    const gradientHeaderTree = renderer
      .create(<Header value="Convenient Transaction" />)
      .toJSON();
    const whiteHeaderTree = renderer
      .create(<Header value="Millionsof people participate" variant="white" />)
      .toJSON();
    expect(gradientHeaderTree).toMatchSnapshot('Gradient Variant');
    expect(whiteHeaderTree).toMatchSnapshot('White Variant');
  });

  test('ensure variants works correctly', () => {
    const { rerender } = render(
      <Header value="Convenient Transaction" testID="atom-header" />,
    );
    let header = screen.getByTestId('atom-header');
    expect(header).toHaveTextContent('Convenient Transaction');
    expect(header).toHaveStyle({ fontSize: 40, fontWeight: '300' });
    rerender(
      <Header
        value="Millionsof people participate"
        variant="white"
        testID="atom-header"
      />,
    );
    header = screen.getByTestId('atom-header');
    expect(header).toHaveTextContent('Millionsof people participate');
    expect(header).toHaveStyle({
      fontSize: 42,
      fontWeight: '400',
      color: '#FFFFFF',
    });
  });
});
