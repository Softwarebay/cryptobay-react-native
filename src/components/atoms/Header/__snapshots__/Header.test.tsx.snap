// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Atom Component Snapshot: Gradient Variant 1`] = `
<RNCMaskedView>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
            "lineHeight": 56,
          },
          Object {
            "fontSize": 40,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      Convenient Transaction
    </Text>
  </View>
  <BVLinearGradient
    colors={
      Array [
        4289318399,
        4285724369,
        4288736653,
        4294890341,
        4294616222,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
            "lineHeight": 56,
          },
          Object {
            "fontSize": 40,
            "fontWeight": "400",
          },
          Object {
            "opacity": 0,
          },
          undefined,
        ]
      }
    >
      Convenient Transaction
    </Text>
  </BVLinearGradient>
</RNCMaskedView>
`;

exports[`Header Atom Component Snapshot: White Variant 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins",
        "lineHeight": 56,
      },
      Object {
        "color": "#FFFFFF",
        "fontSize": 42,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  Millionsof people participate
</Text>
`;
